# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/chromium/src/chrome/common/ipc_message.h
# Commit: c02fa459e77d
# Full Hash: c02fa459e77d2572b2fe740f20d6070a77383589
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-22 21:29:07
# Regressor Bug: 1706374
# File Overlap Count: 2
# Description:
#   Bug 1706374 - Part 13: Remove the event footer from an IPC::Message when deserializing, r=handyman,glandium
#   
#   This unfortunately requires a new method to be added to BufferList to
#   support truncating the buffer to a particular iterator.
#   
# ==============================================================================

diff -r 72dc6537cf0b -r c02fa459e77d ipc/chromium/src/chrome/common/ipc_message.h
--- a/ipc/chromium/src/chrome/common/ipc_message.h	Mon Jun 21 21:53:11 2021 +0000
+++ b/ipc/chromium/src/chrome/common/ipc_message.h	Mon Jun 21 21:53:11 2021 +0000
@@ -282,7 +282,8 @@
   size_t GetSizeIfSerialized() const override { return size(); }
 
   void WriteFooter(const void* data, uint32_t data_len);
-  [[nodiscard]] bool ReadFooter(void* buffer, uint32_t buffer_len);
+  [[nodiscard]] bool ReadFooter(void* buffer, uint32_t buffer_len,
+                                bool truncate);
   uint32_t FooterSize() const;
 
   // Used for async messages with no parameters.